### PR TITLE
Blog scripts

### PR DESCRIPTION
**Pull request by yjose**
Add script to collect last 100 blog posts based on rss feed

### CODE REVIEW

**Review by AbderrahimSoubaiElidrissi — APPROVED** [review]
I think we need to discuss some changes but overall LGTM :+1:

File: .github/workflows/test-scripts.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Install dependencies
+        run: cd ./scripts && yarn install --frozen-lockfile
+
+      - name: Run Tests
+        uses: mattallty/jest-github-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          working-directory: "scripts"
```
[the remark this answers is not in the document]
Comment (yjose) [reply .github/workflows/test-scripts.yml:35]: we are not using it any more `jest-github-action`

File: scripts/src/blog/feed-url-promise.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+type RssItem = Parser.Item;
+type ValidRssItem = Parser.Item & {
+  title: string;
+  link: string;
+  pubDate: string;
+};
+
+// Retry configuration
+const retryConfig = {
```
[the remark this answers is not in the document]
Comment (yjose) [reply scripts/src/blog/feed-url-promise.ts:19]: done

File: scripts/src/blog/feed-url-promise.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      pubDate: new Date(item.pubDate.trim()),
+      image: "",
+      author: "",
+    }))
+    .slice(0, MAX_BLOG_POSTS_BY_RSS_URL);
+
+  return posts as Post[];
+};
+
+function isValidPost(item: RssItem): item is ValidRssItem {
```
[the remark this answers is not in the document]
Comment (yjose) [reply scripts/src/blog/feed-url-promise.ts:55]: Not sure if this is a good idea, i created all those scripts inside the folder blog , maybe i should create a utils inside blog folder

[the remark this answers is not in the document]
Comment (yjose) [reply scripts/src/blog/feed-url-promise.ts:55]: i created a `filters` file for it as  i am going to add more filters depend on rss feed ( meduim and dev.to need some filter logic to collect the correct posts )

File: scripts/src/blog/constants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const EMPTY_LIST_ERROR = `❌ Empty feed List, please check and try again`;
```
[the remark this answers is not in the document]
Comment (yjose) [reply scripts/src/blog/constants.ts]: done

File: scripts/src/blog/generate-posts-from-rss-urls.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      logs: `❌ Empty feed List, please check and try again`,
+    };
+  }
+
+  const promiseArray = rssUrls.map((rssUrl) => feedPromise(rssUrl));
+  const results = await Promise.allSettled(promiseArray);
+
+  const posts: Post[] = [];
+  const errors: string[] = [];
+  results.forEach((result) => {
```
[the remark this answers is not in the document]
Comment (yjose) [reply scripts/src/blog/generate-posts-from-rss-urls.ts]: done

**Pull request merged**